From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jeremy Rose <japthorp@slack-corp.com>
Date: Thu, 23 Jun 2022 16:49:28 -0700
Subject: force disable v8 sandboxed pointers

temporarily, until we can fix node to be compatible

diff --git a/BUILD.gn b/BUILD.gn
index 1b91d1bd0e494bd7b330a0da4280c59e4dd9599f..60359ced28f31540cc6ad706430396155b638ec5 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -506,9 +506,9 @@ if (v8_enable_sandbox == "") {
 }
 
 # Enable sandboxed pointers when the sandbox is enabled.
-if (v8_enable_sandbox) {
-  v8_enable_sandboxed_pointers = true
-}
+#if (v8_enable_sandbox) {
+#  v8_enable_sandboxed_pointers = true
+#}
 
 # Enable all available sandbox features if sandbox future is enabled.
 if (v8_enable_sandbox_future) {
diff --git a/src/api/api.cc b/src/api/api.cc
index 115fda1d2365477dfa1b3902087d063007a5e9aa..dc63b11892644f455f591563938b323fa690c5d0 100644
--- a/src/api/api.cc
+++ b/src/api/api.cc
@@ -7988,14 +7988,14 @@ std::unique_ptr<v8::BackingStore> v8::ArrayBuffer::NewBackingStore(
     void* data, size_t byte_length, v8::BackingStore::DeleterCallback deleter,
     void* deleter_data) {
   CHECK_LE(byte_length, i::JSArrayBuffer::kMaxByteLength);
-#ifdef V8_ENABLE_SANDBOX
+#ifdef V8_SANDBOXED_POINTERS
   Utils::ApiCheck(
       !data || i::GetProcessWideSandbox()->Contains(data),
       "v8_ArrayBuffer_NewBackingStore",
       "When the V8 Sandbox is enabled, ArrayBuffer backing stores must be "
       "allocated inside the sandbox address space. Please use an appropriate "
       "ArrayBuffer::Allocator to allocate these buffers.");
-#endif  // V8_ENABLE_SANDBOX
+#endif  // V8_SANDBOXED_POINTERS
 
   std::unique_ptr<i::BackingStoreBase> backing_store =
       i::BackingStore::WrapAllocation(data, byte_length, deleter, deleter_data,
